src/Microsoft.AspNetCore.Sockets.Client.Http/ServerSentEventsTransport.cs

```diff
@@ -132,6 +137,16 @@ public Task StartAsync(Uri url, Channel<byte[], SendMessage> application, Transf
                 _transportCts.Cancel();
                 stream.Dispose();
                 _logger.ReceiveStopped(_connectionId);
+                pipe.Writer.Complete();
```

Comment:
Is it safe to assume that the stream will end and complete will actually be called? In our tests the wait for `writeTask` hangs because the stream is constantly being written to.